docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/Gam.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic.Trainers.BinaryClassification
+{
+    public static class Gam
+    {
+        // This example requires installation of additional NuGet package
+        // <a href="https://www.nuget.org/packages/Microsoft.ML.FastTree/">Microsoft.ML.FastTree</a>.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+            
+            // Create training and validation datasets.
+            var samples = GenerateData();
+
+            // Convert the dataset to an IDataView.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+
+            // Create train and set set.
+            var dataSets = mlContext.Data.TrainTestSplit(data);
+            var trainSet = dataSets.TrainSet;
+            var validSet = dataSets.TestSet;
+
+            // Create a GAM trainer.
+            // Use a small number of bins for this example.
+            var trainer = mlContext.BinaryClassification.Trainers.Gam(maximumBinCountPerFeature: 16);
+
+            // Fit the model to the data using a validation set.
+            // GAM will use a technique called validation pruning to tune the model after training
+            // to improve generalization.
+            var model = trainer.Fit(trainSet, validSet);
+            
+            // Extract the model parameters.
+            var gam = model.Model.SubModel;
+
+            // Now investigate the bias and shape functions of the GAM model.
+            // The bias represents the average prediction for the training data.
+            Console.WriteLine($"Average prediction: {gam.Bias:0.00}");
+
+            // Let's take a look at the features that the model built. Similar to a linear model, we have
+            // one response per feature. Unlike a linear model, this response is a function instead of a line.
```

Comment:
```suggestion
            // one response per feature. Unlike a linear model, this response is a step function (https://en.wikipedia.org/wiki/Step_function) instead of a line.
```